crates/bevy_ecs/src/entity/mod.rs

```diff
@@ -29,11 +29,17 @@ pub use self::serde::*;
 pub use map_entities::*;
 
 use crate::{archetype::ArchetypeId, storage::SparseSetIndex};
-use std::{
-    convert::TryFrom,
-    fmt, mem,
-    sync::atomic::{AtomicI64, Ordering},
-};
+use std::{convert::TryFrom, fmt, mem, sync::atomic::Ordering};
+
+#[cfg(target_has_atomic = "64")]
+use std::sync::atomic::AtomicI64 as AtomicInt;
+#[cfg(target_has_atomic = "64")]
+type Int = i64;
```

Comment:
Not a huge fan of the name `Int` for this, but using a typedef here makes sense.

Reply:
Yeah, I didn't love it either, some alternatives might be
* `IdCursor` or something like this?
* `NonAtomicInt` - feels a little heavy-handed
* just `int`, although this might get flagged by clippy or something, but is more in line with e.g. `libc` conventions for types like this

Open to other suggestions as well.

Reply:
I prefer `IdCursor` or something to that effect. It's very much used for a single purpose here, and "int" and the like imply a lot more generality.